src/store/index.ts

```diff
@@ -0,0 +1,346 @@
+        server.disconnect()
+      }
+    },
+
+    async fetchUsername(context) {
+      const name = await username()
+      context.commit('setUsername', name)
+    },
+
+    async connect(context, config: SavedConnection) {
+      if (context.state.username) {
+        const server = ConnectionProvider.for(config, context.state.username)
+        // TODO: Check case connection is been created with undefined as key
+        const connection = server.createConnection(config.defaultDatabase || undefined)
+        await connection.connect()
+        connection.connectionType = config.connectionType;
+        const lastUsedConnection = context.state.usedConfigs.find(c => c.hash === config.hash)
+        if (!lastUsedConnection) {
+          const usedConfig = new UsedConnection(config)
+          await usedConfig.save()
+        } else {
+          lastUsedConnection.updatedAt = new Date()
+          if (config.id) {
+            lastUsedConnection.savedConnectionId = config.id
+          }
+          await lastUsedConnection.save()
+        }
+        context.commit('newConnection', {config: config, server, connection})
+      }
+    },
+    async disconnect(context) {
+      const server = context.state.server
+      server?.disconnect()
+      context.commit('clearConnection')
+      context.commit('tables', [])
+    },
+    async changeDatabase(context, newDatabase: string) {
+      if (context.state.server) {
+        const server = context.state.server
+        let connection = server.db(newDatabase)
+        if (! connection) {
+          connection = server.createConnection(newDatabase)
+          await connection.connect()
+        }
+        context.commit('updateConnection', {connection, database: newDatabase})
+        await context.dispatch('updateTables')
+      }
+    },
+    async updateTables(context) {
+      // Ideally here we would run all queries in parallel
+      // however running through an SSH tunnel doesn't work
+      // it only supports one query at a time.
+
+      if (context.state.connection) {
+        try {
+          context.commit("tablesLoading", "Finding tables")
+          const onlyTables = await context.state.connection.listTables({ schema: null })
+          onlyTables.forEach((t) => {
+            t.entityType = 'table'
+          })
+          const views = await context.state.connection.listViews({ schema: null })
+          views.forEach((v) => {
+            v.entityType = 'view'
+          })
+
+          const materialized = await context.state.connection.listMaterializedViews({schema: null})
+          materialized.forEach(v => v.entityType = 'materialized-view')
+          const tables = onlyTables.concat(views).concat(materialized)
+          context.commit("tablesLoading", `Loading ${tables.length} tables`)
+
+          // TODO: I think a the table parameter is missing
+          // @ts-ignore
+          const tableColumns = await context.state.connection.listTableColumns()
```

Comment:
I've coded this to work with or without a table. If no table - it fetches all columns for all tables. So this TODO is fine